samples/boards/esp32/wifi_apsta_mode/README.rst

```diff
@@ -0,0 +1,79 @@
+.. _esp32-wifi-AP-STA-mode-sample:
+
+ESP32 WiFi AP-STA co-existence demo
+###################################
+
+Overview
+********
+
+The WiFi AP-STA mode of the ESP32 allows it to function as both
+an Access Point (AP) and a Station (STA) simultaneously.
+This sample demonstrates how to configure and utilize AP-STA mode.
+
+Configuration and usage of following interfaces is shown in sample.
+
+1. ``AP mode``: AP mode is configured and enabled. DHCPv4 server is also
+   configured to assign IP addresses to the joining station.
+2. ``STA mode``: Provide the SSID and PSK of you router
+
+In this demo, AP-STA mode is enabled using ``CONFIG_ESP32_WIFI_AP_STA_MODE``.
+An additional WiFi node is added in the ``.overlay`` file. The ``net_if``.
+In the sample code, initially, the AP mode is enabled, followed by enabling the STA mode.
+The driver checks if AP mode was previously enabled. If so, it transitions
+the ESP32 into AP-STA mode to support both modes and attempts to connect to the
+AP specified by the provided SSID and PSK.
+
+Requirements
+************
+
+This example should be able to run on any commonly available
+:ref:`esp32_devkitc_wroom` development board without any extra hardware.
+
+To enable or disable ``AP-STA`` mode, modify the ``CONFIG_ESP32_WIFI_AP_STA_MODE`` parameter
+in the ``boards/esp32_devkitc_wroom_procpu.conf`` file of the demo. Moreover, an
+extra WiFi node is included in ``boards/esp32_devkitc_wroom_procpu.overlay``.
+
+By default, AP-STA mode is disabled.
+
+Building, Flashing and Running
+******************************
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/boards/esp32/deep_sleep
+   :board: esp32_devkitc_wroom/esp32/procpu
+   :goals: build flash
+   :compact:
+
+Sample Output
+=================
+
+.. code-block:: console
+
+   *** Booting Zephyr OS build v3.7.0-rc3-104-gd1e5c5b3f9b7 ***
+   [00:00:05.171,000] <inf> MAIN: Turning on AP Mode
+   [00:00:05.172,000] <dbg> net_dhcpv4_server: net_dhcpv4_server_start: Started DHCPv4 server, address pool:
+   [00:00:05.172,000] <dbg> net_dhcpv4_server: net_dhcpv4_server_start:     0: 192.168.4.11
+   [00:00:05.172,000] <dbg> net_dhcpv4_server: net_dhcpv4_server_start:     1: 192.168.4.12
+   [00:00:05.172,000] <dbg> net_dhcpv4_server: net_dhcpv4_server_start:     2: 192.168.4.13
+   [00:00:05.172,000] <dbg> net_dhcpv4_server: net_dhcpv4_server_start:     3: 192.168.4.14
+   [00:00:05.172,000] <inf> MAIN: DHCPv4 server started...
+
+   [00:00:05.348,000] <err> esp32_wifi: Failed to send packet
```

Comment:
How are those error lines related to the demonstration of sample code?

Reply:
I don't know exactly why this error occurs. I thought of it as default behaviour. 

**_On the experience basis:_** This error occurs whenever the device is not connected to any access point. As soon as it gets connected to some Wi-Fi AP it stops occurring. I might be wrong but I think It might be some kind of ping message to test connectivity which fails because of no connectivity. 

Maybe @sylvioalves can help us better understand this error.

Reply:
@zin-HXQ  My point was it is not needed here for demonstration or explanation purposes, so maybe you can simply delete them as they may lead to confusion.